### ISSUE TITLE
input integer #define-s should be changed to enum-s

### ISSUE THREAD

**Issue by samhatchett** [OpenWaterAnalytics/EPANET#30]
... this is mostly a style enhancement which would ease auto-generated documentation by grouping related parameters within a defined enum and provide better support for code-completion in an IDE.

for example:

````
#define EN_CVPIPE       0   
#define EN_PIPE         1    
#define EN_PUMP         2
#define EN_PRV          3
#define EN_PSV          4
#define EN_PBV          5
#define EN_FCV          6
#define EN_TCV          7
#define EN_GPV          8
````

should be converted to:

````
typedef enum {
  EN_CVPIPE       = 0,    
  EN_PIPE         = 1,    
  EN_PUMP         = 2,
  EN_PRV          = 3,
  EN_PSV          = 4,
  EN_PBV          = 5,
  EN_FCV          = 6,
  EN_TCV          = 7,
  EN_GPV          = 8
} EN_LinkType;
````

**Comment by eladsal** [comment]
Is this in `epanet2.h`? Would it break backward compatibility for people who uses these parameters?

**Comment by samhatchett** [comment]
As long as the enums are set up to correspond to the the existing integer values it should work. Then again this may vary depending on the C dialect. I will look into this issue prior to making any changes.

**Comment by samhatchett** [comment]
this ended up being implemented in relation to #33 to ease the documentation of this vocabulary, so i'm closing this to merge into the other issue.
